onnx/hub.py

```diff
@@ -0,0 +1,179 @@
+# SPDX-License-Identifier: Apache-2.0
+
+"""ONNX Model Hub
+
+This implements the python client for the ONNX model hub.
+"""
+from os.path import join
+from urllib.request import urlopen
+from urllib.error import HTTPError
+import json
+import os
+import wget  # type: ignore
+import hashlib
+from io import BytesIO
+from typing import List, Optional, Dict, Any, cast, Set, IO
+import onnx
+
+if "ONNX_HOME" in os.environ:
+    _ONNX_HUB_DIR = join(os.environ["ONNX_HOME"], "hub")
+elif "XDG_CACHE_HOME" in os.environ:
+    _ONNX_HUB_DIR = join(os.environ["XDG_CACHE_HOME"], "onnx", "hub")
+else:
+    _ONNX_HUB_DIR = join(os.path.expanduser('~'), ".cache", "onnx", "hub")
```

Comment:
use %APPDATA% if platform windows?

Reply:
`os.path.expanduser('~')` on Windows already expands to `%userprofile%`. Is there any special reason to use `%APPDATA%` instead?

Reply:
good to know "os.path.expanduser('~') on Windows already expands to %userprofile%" In that case we dont need changes here.